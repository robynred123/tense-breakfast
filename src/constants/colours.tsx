export const BLUE: string = '#041549';
export const TEAL: string = '#35D0BA';
export const GREY: string = '#F3F4F6';
export const DARK_GREY: string = '#374151';
export const WHITE: string = '#FFFF';
