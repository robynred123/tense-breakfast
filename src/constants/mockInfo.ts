export const mockBio: string =
  'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Ut dictum feugiat nisl at condimentum. Pellentesque venenatis augue non ante fringilla euismod. Fusce tempus ultricies malesuada. Fusce eu volutpat nisi. Suspendisse viverra efficitur dui, efficitur rhoncus purus luctus ut. In ut eros maximus, viverra lorem sit amet, pellentesque massa. ';
