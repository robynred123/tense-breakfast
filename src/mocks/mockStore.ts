import configureStore from 'redux-mock-store';
import { Store, AnyAction } from 'redux';
import { FilterOptions, ReducerState } from '../types/index';

export const filterOptions: FilterOptions = {
  appointmentType: [],
  dateRange: {
    start: null,
    end: null,
  },
  specialisms: [],
};
export const initialState: ReducerState = {
  therapists: [],
  filteredTherapists: [],
  filterOptions,
  mobileFilter: false,
  error: null,
};

export const mockStore = (initialState: ReducerState) => {
  const mockStore = configureStore();
  let store: Store<any, AnyAction>;
  store = mockStore({ one: initialState });
  return store;
};
