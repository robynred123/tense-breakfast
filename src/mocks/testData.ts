import { TherapistInfo } from '../types';

export const therapistOne: TherapistInfo = {
  id: '063eb398-45f7-4a9c-a5d3-e95d14c05c44',
  firstName: 'Albert',
  lastName: 'Einstein',
  appointment_types: ['one_off'],
  appointment_mediums: ['phone'],
  specialisms: ['Alcoholism', 'Gender identity', 'Gambling Addiction'],
};

export const therapistTwo: TherapistInfo = {
  id: '154aa58b-4c25-4d64-a7a2-9288a7d94468',
  firstName: 'Jackie',
  lastName: 'Chan',
  appointment_types: ['consultation', 'one_off'],
  appointment_mediums: ['video'],
  specialisms: ['Gender identity', 'Alcoholism'],
};
