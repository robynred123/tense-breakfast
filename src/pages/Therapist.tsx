import { useEffect, useState } from 'react';
import { Circle, Phone, VideoCall } from '@mui/icons-material';
import { Box, Grid, IconButton, MenuItem, Select, Typography } from '@mui/material';
import { useLocation, useNavigate } from 'react-router-dom';
import moment from 'moment';

import { getAvailabilitiesById } from '../actions/actions';
import { ButtonComponent } from '../components/Button';
import { DARK_GREY, TEAL, WHITE } from '../constants/colours';
import { mockBio } from '../constants/mockInfo';
import { AvailabilityData, TherapistInfo, AppointmentType, AppointmentMedium } from '../types';
import { determineSelectedType } from '../utils/filterOptions';

interface State {
  therapist: TherapistInfo;
}

export const TherapistPage = () => {
  const { state } = useLocation();
  const navigate = useNavigate();
  const { therapist } = state as State;
  const [availabilities, setAvailabilities] = useState<AvailabilityData[] | null>(null);
  const [type, setType] = useState<AppointmentType>('one_off');
  const [medium, setMedium] = useState<AppointmentMedium>('phone');
  const [time, setTime] = useState<any>('');

  console.log(time);

  useEffect(() => {
    const asyncFetch = async () => {
      const availabilities = await getAvailabilitiesById(therapist.id);
      return setAvailabilities(
        availabilities.sort((a, b) => Date.parse(a.datetime) - Date.parse(b.datetime))
      );
    };

    asyncFetch();
  }, [therapist]);

  const determineColour = (button: AppointmentMedium, type: string) => {
    if (type === 'text') {
      return medium === button ? 'black' : WHITE;
    } else return medium === button ? TEAL : DARK_GREY;
  };

  return (
    <Box sx={{ flexGrow: 1, width: 1, flexDirection: 'column', justifyContent: 'center' }}>
      <Grid container>
        <Grid container sx={{ flexDirection: 'column' }}>
          {/* Therapist picture/ placeholder */}
          <Grid item sm={4} md={4} lg={4} xl={4}>
            <Circle style={{ color: TEAL, minHeight: '300px', width: '100%' }} />
          </Grid>
          {/* Therapist Name */}
          <Grid
            item
            sm={8}
            md={8}
            lg={8}
            xl={8}
            sx={{ padding: '3em', paddingBottom: 0, width: '100%' }}
          >
            <Typography
              variant='h2'
              fontFamily={'lato, sans-serif'}
              sx={{ justifyContent: 'flex-start' }}
            >
              {therapist.firstName} {therapist.lastName}
            </Typography>
            {/* Bio section */}
            <Typography
              variant='body1'
              fontFamily={'lato, sans-serif'}
              sx={{ justifyContent: 'flex-start' }}
            >
              {mockBio}
            </Typography>

            {/* Appointment Time */}
            <Grid container sx={{ paddingTop: '3em', flexDirection: 'column' }}>
              <Grid item sm={12} md={12} lg={12} xl={12}>
                <Typography
                  variant='h5'
                  fontFamily={'lato, sans-serif'}
                  sx={{ justifyContent: 'flex-start' }}
                >
                  Appointment Time
                </Typography>

                <Grid item sx={{ justifyContent: 'space-evenly', display: 'flex' }}>
                  <Select
                    value={time}
                    onChange={(e) => setTime(e.target.value)}
                    sx={{ width: '90%' }}
                  >
                    {availabilities ? (
                      availabilities.map((appointment) => {
                        return (
                          <MenuItem value={appointment.datetime} key={appointment.id}>
                            {moment(appointment.datetime).format('LLLL')}
                          </MenuItem>
                        );
                      })
                    ) : (
                      <p>No Appointments Available</p>
                    )}
                  </Select>
                </Grid>
              </Grid>
            </Grid>
            {/* Appointment Type */}
            <Grid container sx={{ paddingTop: '3em', flexDirection: 'column' }}>
              <Grid item sm={12} md={12} lg={12} xl={12}>
                <Typography
                  variant='h5'
                  fontFamily={'lato, sans-serif'}
                  sx={{ justifyContent: 'flex-start' }}
                >
                  Appointment Type
                </Typography>

                <Grid item sx={{ justifyContent: 'space-evenly', display: 'flex' }}>
                  <ButtonComponent
                    onClick={() => setType('one_off')}
                    text='One Off'
                    buttonColour={determineSelectedType('one_off', [type])}
                    disabled={!therapist.appointment_types.includes('one_off')}
                    width={'40%'}
                  />
                  <ButtonComponent
                    onClick={() => setType('consultation')}
                    text='Consultation'
                    buttonColour={determineSelectedType('consultation', [type])}
                    disabled={!therapist.appointment_types.includes('consultation')}
                    width={'40%'}
                  />
                </Grid>
              </Grid>

              {/* Appointment Method */}
              <Grid container sx={{ paddingTop: '3em', flexDirection: 'column' }}>
                <Grid item sm={12} md={12} lg={12} xl={12}>
                  <Typography
                    variant='h5'
                    fontFamily={'lato, sans-serif'}
                    sx={{ justifyContent: 'flex-start' }}
                  >
                    Appointment Medium
                  </Typography>

                  <Grid item sx={{ justifyContent: 'space-evenly', display: 'flex' }}>
                    <IconButton
                      style={{
                        minHeight: '80px',
                        width: '40%',
                        backgroundColor: determineColour('phone', 'button'),
                        borderRadius: 10,
                        flexDirection: 'column',
                      }}
                      disabled={!therapist.appointment_mediums.includes('phone')}
                      onClick={() => setMedium('phone')}
                    >
                      <Phone
                        style={{
                          color: determineColour('phone', 'text'),
                          minHeight: '90%',
                          width: '100%',
                          minWidth: '75px',
                          padding: 0,
                        }}
                      />
                      <Typography
                        color={determineColour('phone', 'text')}
                        fontFamily={'lato, sans-serif'}
                      >
                        Phone
                      </Typography>
                    </IconButton>

                    <IconButton
                      style={{
                        minHeight: '80px',
                        width: '40%',
                        backgroundColor: determineColour('video', 'button'),
                        borderRadius: 10,
                        flexDirection: 'column',
                      }}
                      disabled={!therapist.appointment_mediums.includes('video')}
                      onClick={() => setMedium('video')}
                    >
                      <VideoCall
                        style={{
                          color: determineColour('video', 'text'),
                          minHeight: '90%',
                          width: '100%',
                          minWidth: '75px',
                          padding: 0,
                        }}
                      />
                      <Typography
                        color={determineColour('video', 'text')}
                        fontFamily={'lato, sans-serif'}
                      >
                        Video
                      </Typography>
                    </IconButton>
                  </Grid>
                </Grid>
                {/* Submit and Cancel buttons */}
                <Grid
                  container
                  sx={{
                    paddingTop: '3em',
                    flexDirection: 'row',
                    justifyContent: 'space-evenly',
                    alignContent: 'flex-end',
                  }}
                >
                  <ButtonComponent
                    onClick={() => navigate(-1)}
                    text='Cancel'
                    buttonColour={'grey'}
                    disabled={false}
                    width={'40%'}
                  />

                  <ButtonComponent
                    onClick={() => console.log('submit')}
                    text='Submit'
                    buttonColour={'green'}
                    disabled={false}
                    width={'40%'}
                  />
                </Grid>
              </Grid>
            </Grid>
          </Grid>
        </Grid>
      </Grid>
    </Box>
  );
};
