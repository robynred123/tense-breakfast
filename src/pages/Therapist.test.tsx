/* eslint-disable testing-library/await-async-query */
import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { therapistOne } from '../mocks/testData';
import { mockStore, initialState } from '../mocks/mockStore';
import { Router } from 'react-router';
import { TherapistPage } from './Therapist';
import { createBrowserHistory } from 'history';
import { mockBio } from '../constants/mockInfo';

describe('<TherapistPage />', () => {
  let mockUseLocationValue = {
    pathname: '/Therapist',
    key: '',
    search: '',
    hash: '',
    state: { therapist: therapistOne },
  };

  const history = createBrowserHistory();

  beforeEach(() => {
    jest.resetAllMocks();
  });

  it('Should render the Therapist page successfully', () => {
    const store = mockStore(initialState);
    const view = renderer.create(
      <Provider store={store}>
        <Router navigator={history} location={mockUseLocationValue}>
          <TherapistPage />
        </Router>
      </Provider>
    );

    expect(view.toJSON()).toMatchSnapshot();
    // Therapist Data
    expect(view.root.findAllByType('svg')[0].props['data-testid']).toEqual('CircleIcon');
    expect(view.root.findByType('h2').props.children).toEqual(['Albert', ' ', 'Einstein']);
    expect(view.root.findAllByType('p')[0].props.children).toEqual(mockBio);

    // Date selector
    expect(view.root.findAllByType('h5')[0].props.children).toEqual('Appointment Time');
    expect(view.root.findByProps({ 'aria-haspopup': 'listbox' })).toBeDefined();
    expect(view.root.findAllByType('input')[0].props.className).toEqual(
      'MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput'
    );

    // Appointment Type
    expect(view.root.findAllByType('h5')[1].props.children).toEqual('Appointment Type');
    expect(view.root.findAllByType('button')[0].props.children).toEqual([
      [undefined, 'One Off', undefined],
      null,
    ]);
    expect(view.root.findAllByType('button')[0].props.disabled).toEqual(false);
    expect(view.root.findAllByType('button')[1].props.children).toEqual([
      [undefined, 'Consultation', undefined],
      null,
    ]);
    expect(view.root.findAllByType('button')[1].props.disabled).toEqual(true);

    // Appointment Medium
    expect(view.root.findAllByType('h5')[2].props.children).toEqual('Appointment Medium');
    const phoneSvg = view.root.findAllByType('svg')[2];
    expect(phoneSvg.props['data-testid']).toEqual('PhoneIcon');
    expect(view.root.findAllByType('button')[2].props.disabled).toEqual(false);
    expect(view.root.findAllByType('p')[1].props.children).toEqual('Phone');
    const videoSvg = view.root.findAllByType('svg')[3];
    expect(videoSvg.props['data-testid']).toEqual('VideoCallIcon');
    expect(view.root.findAllByType('button')[3].props.disabled).toEqual(true);
    expect(view.root.findAllByType('p')[2].props.children).toEqual('Video');

    // End Buttons
    expect(view.root.findAllByType('button')[4].props.disabled).toEqual(false);
    expect(view.root.findAllByType('button')[4].props.children).toEqual([
      [undefined, 'Cancel', undefined],
      null,
    ]);
    expect(view.root.findAllByType('button')[5].props.disabled).toEqual(true);
    expect(view.root.findAllByType('button')[5].props.children).toEqual([
      [undefined, 'Book', undefined],
      null,
    ]);
  });
});
