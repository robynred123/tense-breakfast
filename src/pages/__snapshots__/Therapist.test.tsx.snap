// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TherapistPage /> Should render the Therapist page successfully 1`] = `
<div
  className="MuiBox-root css-rjqiu3"
>
  <div
    className="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
  >
    <div
      className="MuiGrid-root MuiGrid-container css-k6wrnl-MuiGrid-root"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-4 MuiGrid-grid-md-4 MuiGrid-grid-lg-4 MuiGrid-grid-xl-4 css-kgs8gf-MuiGrid-root"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="CircleIcon"
          focusable="false"
          style={
            Object {
              "color": "#35D0BA",
              "minHeight": "300px",
              "width": "100%",
            }
          }
          viewBox="0 0 24 24"
        >
          <path
            d="M12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2z"
          />
        </svg>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-8 MuiGrid-grid-md-8 MuiGrid-grid-lg-8 MuiGrid-grid-xl-8 css-nzahwq-MuiGrid-root"
      >
        <h2
          className="MuiTypography-root MuiTypography-h2 css-18neeym-MuiTypography-root"
        >
          Albert
           
          Einstein
        </h2>
        <p
          className="MuiTypography-root MuiTypography-body1 css-efofzw-MuiTypography-root"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Ut dictum feugiat nisl at condimentum. Pellentesque venenatis augue non ante fringilla euismod. Fusce tempus ultricies malesuada. Fusce eu volutpat nisi. Suspendisse viverra efficitur dui, efficitur rhoncus purus luctus ut. In ut eros maximus, viverra lorem sit amet, pellentesque massa. 
        </p>
        <div
          className="MuiGrid-root MuiGrid-container css-ro30o1-MuiGrid-root"
        >
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-12 MuiGrid-grid-md-12 MuiGrid-grid-lg-12 MuiGrid-grid-xl-12 css-13shxg4-MuiGrid-root"
          >
            <h5
              className="MuiTypography-root MuiTypography-h5 css-x242tx-MuiTypography-root"
            >
              Appointment Time
            </h5>
            <div
              className="MuiGrid-root MuiGrid-item css-1urxl9a-MuiGrid-root"
            >
              <div
                className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary  css-t8rt5h-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
                onClick={[Function]}
              >
                <div
                  aria-expanded="false"
                  aria-haspopup="listbox"
                  className="MuiSelect-select MuiSelect-outlined MuiOutlinedInput-input MuiInputBase-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
                  onBlur={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onMouseDown={[Function]}
                  role="button"
                  tabIndex={0}
                >
                  <span
                    className="notranslate"
                  >
                    ​
                  </span>
                </div>
                <input
                  aria-hidden={true}
                  className="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                  onAnimationStart={[Function]}
                  onChange={[Function]}
                  tabIndex={-1}
                  value=""
                />
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
                  data-testid="ArrowDropDownIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
                <fieldset
                  aria-hidden={true}
                  className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    className="css-hdw1oc"
                  >
                    <span
                      className="notranslate"
                    >
                      ​
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-container css-ro30o1-MuiGrid-root"
        >
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-12 MuiGrid-grid-md-12 MuiGrid-grid-lg-12 MuiGrid-grid-xl-12 css-13shxg4-MuiGrid-root"
          >
            <h5
              className="MuiTypography-root MuiTypography-h5 css-x242tx-MuiTypography-root"
            >
              Appointment Type
            </h5>
            <div
              className="MuiGrid-root MuiGrid-item css-1urxl9a-MuiGrid-root"
            >
              <button
                className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1d5f0aw-MuiButtonBase-root-MuiButton-root"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onContextMenu={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={0}
                type="button"
              >
                One Off
              </button>
              <button
                className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root Mui-disabled  css-1d5f0aw-MuiButtonBase-root-MuiButton-root"
                disabled={true}
                onBlur={[Function]}
                onClick={[Function]}
                onContextMenu={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={-1}
                type="button"
              >
                Consultation
              </button>
            </div>
          </div>
          <div
            className="MuiGrid-root MuiGrid-container css-ro30o1-MuiGrid-root"
          >
            <div
              className="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-12 MuiGrid-grid-md-12 MuiGrid-grid-lg-12 MuiGrid-grid-xl-12 css-13shxg4-MuiGrid-root"
            >
              <h5
                className="MuiTypography-root MuiTypography-h5 css-x242tx-MuiTypography-root"
              >
                Appointment Medium
              </h5>
              <div
                className="MuiGrid-root MuiGrid-item css-1urxl9a-MuiGrid-root"
              >
                <button
                  className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                  disabled={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onContextMenu={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  style={
                    Object {
                      "backgroundColor": "#374151",
                      "borderRadius": 10,
                      "flexDirection": "column",
                      "minHeight": "80px",
                      "width": "40%",
                    }
                  }
                  tabIndex={0}
                  type="button"
                >
                  <svg
                    aria-hidden={true}
                    className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="PhoneIcon"
                    focusable="false"
                    style={
                      Object {
                        "color": "#FFFF",
                        "minHeight": "90%",
                        "minWidth": "75px",
                        "padding": 0,
                        "width": "100%",
                      }
                    }
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M6.62 10.79c1.44 2.83 3.76 5.14 6.59 6.59l2.2-2.2c.27-.27.67-.36 1.02-.24 1.12.37 2.33.57 3.57.57.55 0 1 .45 1 1V20c0 .55-.45 1-1 1-9.39 0-17-7.61-17-17 0-.55.45-1 1-1h3.5c.55 0 1 .45 1 1 0 1.25.2 2.45.57 3.57.11.35.03.74-.25 1.02l-2.2 2.2z"
                    />
                  </svg>
                  <p
                    className="MuiTypography-root MuiTypography-body1 css-ags0sf-MuiTypography-root"
                  >
                    Phone
                  </p>
                </button>
                <button
                  className="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                  disabled={true}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onContextMenu={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  style={
                    Object {
                      "backgroundColor": "#374151",
                      "borderRadius": 10,
                      "flexDirection": "column",
                      "minHeight": "80px",
                      "width": "40%",
                    }
                  }
                  tabIndex={-1}
                  type="button"
                >
                  <svg
                    aria-hidden={true}
                    className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="VideoCallIcon"
                    focusable="false"
                    style={
                      Object {
                        "color": "#FFFF",
                        "minHeight": "90%",
                        "minWidth": "75px",
                        "padding": 0,
                        "width": "100%",
                      }
                    }
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M17 10.5V7c0-.55-.45-1-1-1H4c-.55 0-1 .45-1 1v10c0 .55.45 1 1 1h12c.55 0 1-.45 1-1v-3.5l4 4v-11l-4 4zM14 13h-3v3H9v-3H6v-2h3V8h2v3h3v2z"
                    />
                  </svg>
                  <p
                    className="MuiTypography-root MuiTypography-body1 css-ags0sf-MuiTypography-root"
                  >
                    Video
                  </p>
                </button>
              </div>
            </div>
            <div
              className="MuiGrid-root MuiGrid-container css-1idcksl-MuiGrid-root"
            >
              <button
                className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1d5f0aw-MuiButtonBase-root-MuiButton-root"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onContextMenu={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={0}
                type="button"
              >
                Cancel
              </button>
              <button
                className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root Mui-disabled  css-1ehb03l-MuiButtonBase-root-MuiButton-root"
                disabled={true}
                onBlur={[Function]}
                onClick={[Function]}
                onContextMenu={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={-1}
                type="button"
              >
                Book
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
